test/admin/authProviders.test.js

```diff
@@ -0,0 +1,97 @@
+const StitchMongoFixture = require('../fixtures/stitch_mongo_fixture');
+
+import {getAuthenticatedClient} from '../testutil';
+
+describe('Auth Providers V2', ()=>{
+  let test = new StitchMongoFixture();
+  let authProviders;
+  let app;
+  let apps;
+  let validConfig = {
+    emailConfirmationUrl: 'http://emailConfirmURL.com',
+    resetPasswordUrl: 'http://resetPasswordURL.com',
+    confirmEmailSubject: 'email subject',
+    resetPasswordSubject: 'password subject'
+  };
+  beforeAll(() => test.setup({createApp: false}));
+  afterAll(() => test.teardown());
+  beforeEach(async () =>{
+    let adminClient = await getAuthenticatedClient(test.userData.apiKey.key);
+    test.groupId = test.userData.group.groupId;
+    apps = await adminClient.v2().apps(test.groupId);
+    app = await apps.create({name: 'testname'});
+    authProviders = adminClient.v2().apps(test.groupId).app(app._id).authProviders();
+  });
+  afterEach(async () => {
+    await apps.app(app._id).remove();
+  });
+
+  it('listing auth providers should return empty list', async () => {
+    let providers = await authProviders.list();
+    expect(providers).toEqual([]);
+  });
+  it('creating auth provider should work', async () => {
+    let newProvider = await authProviders.create({type: 'local-userpass', config: validConfig});
+    expect(newProvider.type).toEqual('local-userpass');
+    expect(newProvider.name).toEqual('local-userpass');
+    expect(newProvider.config).toEqual(undefined);
+    let providers = await authProviders.list();
+    expect(providers).toHaveLength(1);
+    expect(providers[0].type).toEqual(newProvider.type);
+  });
+  it('invalid create requests should fail', async () => {
+    await expect(authProviders.create({type: 'local-userpass'})).rejects.toBeDefined();
+    await expect(authProviders.create({type: 'bad#type'})).rejects.toBeDefined();
+  });
+  it('fetching auth provider should work', async () => {
+    let newAuthProvider = await authProviders.create({type: 'local-userpass', config: validConfig});
+    expect(newAuthProvider.type).toEqual('local-userpass');
+    let provider = await authProviders.authProvider(newAuthProvider._id).get();
+    expect(provider.type).toEqual(provider.type);
+  });
+  it('enabling auth provider should work', async () => {
+    let newAuthProvider = await authProviders.create({type: 'local-userpass', config: validConfig, disabled: true});
+    expect(newAuthProvider.type).toEqual('local-userpass');
```

Comment:
It might be worth doing an additional assertion here to verify that `disabled` is actually `true` for the newly created key, so we know that the flag is actually effective when used as part of the `create()` method, and that the `enable()` call actually had an effect.
Similar for the next test too, after L61. We should check that `disabled` is `false`, before proceeding to call `disable` and verifying its transition to `true`.